File: idcs-sdk-sample-apps/NET/SampleApp/SampleApp/Models/ConnectionOptions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SampleApp.Models
{
    public class ConnectionOptions
    {
        private System.Collections.Generic.Dictionary<String, String> options = null;

        public Dictionary<string, string> GetOptions()
        {
            this.options = new System.Collections.Generic.Dictionary<String, String>
            {
                { "ClientId", "123456789abcdefghij" },
                { "ClientSecret", "abcde-12345-zyxvu-98765-qwerty" },
                { "BaseUrl", "https://idcs-abcd1234.identity.oraclecloud.com" },
                { "AudienceServiceUrl", "https://idcs-abcd1234.identity.oraclecloud.com" },
                { "TokenIssuer", "https://identity.oraclecloud.com/" },
                { "scope", "urn:opc:idm:t.user.me openid" },
                { "redirectURL", "http://localhost:3001/Home/Callback" },
                { "logoutSufix", "/oauth2/v1/userlogout"},
                { "LogLevel", "0" },
                { "ConsoleLog", "True" }
            };
            return this.options;
        }

    }

    
}

File: idcs-sdk-sample-apps/java/src/main/java/sampleapp/util/ConnectionOptions.java
/*
 *  Copyright (c) 2000, 2021, Oracle and/or its affiliates.
 *
 *   Licensed under the Universal Permissive License v 1.0 as shown at
 *   http://oss.oracle.com/licenses/upl.
 */
package sampleapp.util;

import java.util.HashMap;
import java.util.Map;
import oracle.security.jps.idcsbinding.shared.Constants;
import oracle.security.jps.idcsbinding.shared.IDCSTokenAssertionConfiguration;


/**
 * This class provides the parameters for the SDK to connect with a 
 * Oracle Identity Cloud Service instance.
 * The list of parameters must be added to a HashMap.
 * @Copyright Oracle
 */
public class ConnectionOptions {
    
    //Instance of a HashMap.
    private Map<String,Object> options = new HashMap<>();
    
    public ConnectionOptions(){
        this.options = new HashMap<>();
    }
    
    public Map<String,Object> getOptions(){
        //Adding Oracle Identity Cloud Service connection parameters to the HashMap instance.
        this.options.put(IDCSTokenAssertionConfiguration.IDCS_HOST, "identity.oraclecloud.com");
        this.options.put(IDCSTokenAssertionConfiguration.IDCS_PORT, "443");
        this.options.put(IDCSTokenAssertionConfiguration.IDCS_CLIENT_ID, "123456789abcdefghij");
        this.options.put(IDCSTokenAssertionConfiguration.IDCS_CLIENT_SECRET, "abcde-12345-zyxvu-98765-qwerty");
        this.options.put(IDCSTokenAssertionConfiguration.IDCS_CLIENT_TENANT, "idcs-abcd1234");
        this.options.put(Constants.AUDIENCE_SERVICE_URL, "https://idcs-abcd1234.identity.oraclecloud.com");
        this.options.put(Constants.TOKEN_ISSUER, "https://identity.oraclecloud.com/");
        this.options.put(Constants.TOKEN_CLAIM_SCOPE, "urn:opc:idm:t.user.me openid");
        this.options.put("SSLEnabled", "true");
        this.options.put("redirectURL", "http://localhost:8080/callback");
        this.options.put("logoutSufix", "/oauth2/v1/userlogout");
        this.options.put(Constants.LOG_LEVEL, "INFO");
        this.options.put(Constants.CONSOLE_LOG, "True");
        return this.options;
    }
}
